vj4/model/adaptor/setting.py

```diff
@@ -61,11 +75,13 @@ def get_setting(self, key):
       return next(iter(setting.range))
     return setting.factory()
 
-  async def set_settings(self, **kwargs):
+  async def set_settings(self, category, **kwargs):
```

Comment:
get_setting and set_settings should be symmetric in some way - should not accept category in set_settings but not in get_setting. I think you can just deal with individual settings and ignore category here.